mapswipe_workers/tests/integration/test_firebase_db_rules.py

```diff
@@ -0,0 +1,329 @@
+import json
+import unittest
+from requests.exceptions import HTTPError
+import requests
+import re
+from firebase_admin.auth import UserRecord
+from mapswipe_workers.config import FIREBASE_DB, FIREBASE_API_KEY
+from mapswipe_workers.auth import firebaseDB
+from mapswipe_workers.definitions import logger, CustomError
+from mapswipe_workers.utils import user_management
+
+
+def set_up_team(team_id, team_name, team_token):
+    fb_db = firebaseDB()
+    data = {"teamName": team_name, "teamToken": team_token}
+    ref = fb_db.reference(f"v2/teams/{team_id}")
+    ref.set(data)
+
+
+def tear_down_team(team_id: str):
+    fb_db = firebaseDB()
+    # check if reference path is valid, e.g. if team_id is None
+    ref = fb_db.reference(f"v2/teams/{team_id}")
+    if not re.match(r"/v2/\w+/[-a-zA-Z0-9]+", ref.path):
+        raise CustomError(
+            f"""Given argument resulted in invalid Firebase Realtime Database reference.
+                                    {ref.path}"""
+        )
+
+    # delete team in firebase
+    ref.delete()
+
+
+def tear_down_project(project_id: str):
+    fb_db = firebaseDB()
+    # check if reference path is valid, e.g. if team_id is None
+    ref = fb_db.reference(f"v2/projects/{project_id}")
+    if not re.match(r"/v2/\w+/[-a-zA-Z0-9]+", ref.path):
+        raise CustomError(
+            f"""Given argument resulted in invalid Firebase Realtime Database reference.
+                                    {ref.path}"""
+        )
+
+    # delete team in firebase
+    ref.delete()
+
+
+def setup_user(
+    project_manager: bool, team_member: bool, team_id: str = ""
+) -> UserRecord:
+    if project_manager and team_member:
+        username = f"unittest-project-manager-and-team-member"
+    elif project_manager:
+        username = f"unittest-project-manager"
+    elif team_member:
+        username = f"unittest-team-member"
+    else:
+        username = f"unittest-normal-user"
+
+    email = f"{username}@mapswipe.org"
+    password = f"{username}_pw"
+
+    # username will be user.display_name
+    user = user_management.create_user(email, username, password)
+
+    # set project manager credentials
+    if project_manager:
+        user_management.set_project_manager_rights(email)
+
+    # set team member attribute
+    if team_member:
+        user_management.add_user_to_team(email, team_id)
+
+    return user
+
+
+def sign_in_with_email_and_password(email: str, password: str) -> dict:
```

Comment:
I think the return value should be an uppercase `Dict` instead of `dict`. I'm not sure what happens with the lowercase version, I can't find any docs about it. The [mypy official cheatseet](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html#built-in-types) uses the uppercase version though.